rudder-web/src/main/scala/com/normation/rudder/web/components/RuleEditForm.scala

```diff
@@ -227,10 +227,14 @@ class RuleEditForm(
             $$( "#editRuleZone" ).tabs('select', ${tab});"""
           )) &
           JsRaw(s"""
-
             $$("#editRuleZonePortlet").removeClass("nodisplay");
             ${Replace("details", new RuleCompliance(rule, rootRuleCategory).display).toJsCmd};
             scrollToElement("${idToScroll}", ".rudder_col");
+            $$("#editRuleZone").bind( "show", function(event, ui) {
+              if(ui.panel.id== 'ruleComplianceTab') {
+                recentChart.flush();
```

Comment:
Before flushing, you should check that recentChart is not undefined, unless you will have a js error when calling flush on 'undefined'